docs/src/main/tut/testing.md

```diff
@@ -0,0 +1,155 @@
+---
+menu: main
+weight: 350
+title: Testing
+---
+
+## Introduction
+
+This document will show a simple Swagger Doc, its implementation as an `org.http4s.HttpService`, and then
+walk through the results of applying inputs, i.e. `org.http4s.Request`, to the Service, i.e. `org.http4s.HttpService`.
+
+After reading this doc, the reader should feel comfortable writing a unit test using his/her favorite Scala testing library.
+
+## Example
+
+Given the following service, which we can define through the following [Swagger](https://swagger.io/) contract:
+
+```yaml
+swagger: "2.0"
+info:
+  description: "Example Swagger Doc"
+  version: "1.0.0"
+  title: "Http4s Testing Demo"
+host: "localhost"
+basePath: "/api"
+schemes:
+- "http"
+paths:
+  /user/{id}:
+    get:
+      parameters:
+      - name: "id"
+        in: "path"
+        description: "ID of User."
+        required: true
+        type: "string"    
+      summary: "Get a user by id."
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "Got representation of User."
+          schema:
+            $ref: "#/definitions/User"
+        404: 
+          description: "User does not exist."
+        500:
+          description: "Server-side error."  
+definitions:
+  User:
+    type: "object"
+    required:
+    - "name"
+    - "age"
+    properties:
+      name:
+        type: "string"
+      age:
+        type: "integer"
+``` 
+
+Now, let's define the `org.http4s.HttpService`. It will implement the above contract.
+
+```tut:book
+import cats.implicits._
+import io.circe._
+import io.circe.syntax._
+import io.circe.generic.semiauto._
+import cats.effect._, org.http4s._, org.http4s.dsl.io._
+import org.http4s.circe._
+import scala.util.Try
+
+final case class User(name: String, age: Int)
+
+implicit val UserEncoder: Encoder[User] = deriveEncoder[User]
```

Comment:
Should this top level `val` be here? It doesn't compile, does it?

Reply:
Ah, it will compile as in a REPL session, but not on "normal" code. This is a bit confusing.

Reply:
I originally put the `Encoder[User]` in `object User`, i.e. the companion. However, if I recall correctly, I got a `tut` error due to, as I understand, the `implicit` not being in scope, i.e. due to the REPL.

Please let me know what you suggest, @jcranky, to make it clear!

Reply:
Worst case:

Put the implicit val in the companion obj, then `import User._ //note: this is necessary for the repl`.

Pretty much all you need.

Reply:
@jmcardon 's suggestion would be enough. Other than that, I recall that there were some trick to make that work with repl, like the `:paste` mode, but I don't know how to express that for `tut`